addon/components/frost-list.js

```diff
@@ -202,7 +224,21 @@ export default Component.extend({
       } else {
         selection.basic(clonedSelectedItems, item, _rangeState, _itemComparator)
       }
-      this.onSelectionChange(clonedSelectedItems)
+
+      const itemTypes = this.get('itemTypes')
+      const itemTypeKey = this.get('itemTypeKey')
+
+      const selectedTypesWithControls = clonedSelectedItems.reduce((typesWithControls, item) => {
+        if (isPresent(itemTypes) && isPresent(itemTypeKey)) {
```

Comment:
I don't think we need to check this on every loop, if it's not defined on line 230 it will be not defined for all loop iteration.

Reply:
I have moved the if check outside of the iteration. I think the original reason why I did it that way was so that I would never be passing `undefined` to the `onSelectionChange` callback lower down - because the `reduce` would always provide an initial value even if there were no iterations.